horovod/spark/common/store.py

```diff
@@ -275,66 +262,83 @@ def _get_full_path_or_default(self, path, default_key):
     def _get_path(self, key):
         return os.path.join(self.prefix_path, key)
 
-    def path_prefix(self):
-        raise NotImplementedError()
+    def get_local_output_dir_fn(self, run_id):
+        @contextlib.contextmanager
+        def local_run_path():
+            with tempfile.TemporaryDirectory() as tmpdir:
+                yield tmpdir
+        return local_run_path
 
-    def get_filesystem(self):
+    def get_localized_path(self, path):
         raise NotImplementedError()
 
-    @classmethod
-    def matches(cls, path):
-        return path.startswith(cls.filesystem_prefix())
+    def get_full_path(self, path):
+        raise NotImplementedError()
 
-    @classmethod
-    def filesystem_prefix(cls):
+    def get_full_path_fn(self):
         raise NotImplementedError()
 
+    @property
+    def fs(self):
+        raise NotImplementedError()
 
-class LocalStore(FilesystemStore):
-    """Uses the local filesystem as a store of intermediate data and training artifacts."""
 
-    FS_PREFIX = 'file://'
+class FilesystemStore(AbstractFilesystemStore):
+    """Concrete filesystems store that delegates to `fsspec`."""
 
     def __init__(self, prefix_path, *args, **kwargs):
-        self._fs = pa.LocalFileSystem()
-        super(LocalStore, self).__init__(prefix_path, *args, **kwargs)
+        self._fs, self.protocol = FilesystemStore._get_fs_and_protocol(prefix_path)
```

Comment:
Can this be added ?:
self.storage_options = kwargs['storage_options'] if 'storage_options' in kwargs else dict()
self._fs, self.protocol = FilesystemStore._get_fs_and_protocol(prefix_path, self.storage_options)